src/docs/binding-overview.md

```diff
@@ -0,0 +1,105 @@
+# Overview of Binding
+
+Binding is the process of making Swift code available to C#. This means being able to access Swift types from C# as well as being able to make a set of C# types available to Swift. In this process there are a number of things that need to be taken into consideration:
+- Language parity mismatches
+- ABI (application binary interface) mismatches
+- Runtime differences
+- Idiomatic differences
+- Accessibility
+
+## Language Parity Mismatches
+Swift is a hybrid programming language leaning towards functional. It has all the features that you would expect in a traditional object oriented programming language, but from Apple's point of view, these are not features that are encouraged. Instead, Apple is encouraging using value types for most everything using copy-on-write semantics for the collection types. In fact, the main runtime for swift, swiftCore, contains 0 published classes as of a few years ago. In addition, Apple encourages "aspect oriented" rather than object oriented coding.
+
+In addition, Swift has a number of language features that do not exist in C#:
+- Discriminated Unions - Swift enums are build so that any case in the set may have payloads and the payloads need not be the same type
```

Comment:
```suggestion
- Discriminated Unions - Swift enums are built so that any case in the set may have payloads and the payloads need not be the same type.
```